hyperspy/signal.py

```diff
@@ -3094,53 +3127,32 @@ def get_dynamic_image_explorer(*args, **kwargs):
                         s=self,
                         axis=self.axes_manager.signal_axes,
                     )
-                if navigator.axes_manager.navigation_dimension == 1:
-                    navigator = interactive(
-                        f=navigator.as_signal1D,
-                        event=navigator.events.data_changed,
-                        recompute_out_event=navigator.axes_manager.events.any_axis_changed,
-                        spectral_axis=0,
-                    )
-                else:
-                    navigator = interactive(
-                        f=navigator.as_signal2D,
-                        event=navigator.events.data_changed,
-                        recompute_out_event=navigator.axes_manager.events.any_axis_changed,
-                        image_axes=(0, 1),
-                    )
             else:
                 navigator = None
         # Navigator properties
         if axes_manager.navigation_axes:
             # check first if we have a signal to avoid comparison of signal with
             # string
             if isinstance(navigator, BaseSignal):
-
-                def is_shape_compatible(navigation_shape, shape):
-                    return (
-                        navigation_shape == shape
-                        or navigation_shape[:2] == shape
-                        or (navigation_shape[0],) == shape
+                if navigator.axes_manager.signal_dimension > 0:
```

Comment:
Is it going to be weird if we deprecate this and then bring this back at some point? 

I __would__ like to support navigators with signal dimensions eventually.

Reply:
I can have a look: It may possible without too much trouble. Can you elaborate how do you see working?
Something as:
```python
s = hs.signals.Signal1D(np.arange(5*10*20*40).reshape((20, 10, 5, 40)))
s2 = s.sum(-1).as_signal1D(-1)
s.plot(navigator=s2)
```
Where the `s` pointer will also be added to the signal dimension of `s2`?

As we need to check the consistent in shape between the signal dimension of the provided navigator signal (`s2`) and the corresponding navigation dimension(s) of `s`, how do we define which navigation dimension of `s` should it be matched to?

Reply:
Okay so I'll start with the example that is most relevant to me:

If you have a time resolved 4D STEM dataset you might want something like this:


```python
updating_nav_pos = np.empty(5, dtype=object)
for i in np.ndindex(updating_nav_pos.shape):
    updating_nav_pos[i] = np.random.random((np.random.randint(3,5), 2))
p1 = hs.plot.markers.Points(updating_nav_pos)

full5Ddata.plot(navigator=time_series_navigator) # plots three figures the navigator for updating_nav_pos, the signal for updating_nav_pos and the signal for full5Ddata 
time_series_navigator.add_marker(p1) # add an updating navigator to the navigator plot

```

With the output being something like:

Navigator 1 &2:
<img width="1133" alt="image" src="https://github.com/hyperspy/hyperspy/assets/41125831/980371b4-42cf-4dd7-b790-e9e7b86b3987">

Signal: 
<img width="587" alt="image" src="https://github.com/hyperspy/hyperspy/assets/41125831/b4c2fdfd-4cd9-4bf9-9f99-7a858dd0d968">

Reply:
This shouldn't be too difficult from the standpoint of writing new code, but tracking everything is going be a bit of a pain so I might end up coming back to it in a bit once I get the time.

Reply:
@CSSFrancis, I removed the deprecation.